playbooks/common/openshift-cluster/validate_hostnames.yml

```diff
@@ -0,0 +1,28 @@
+---
+- include: evaluate_groups.yml
+
+- name: Gather and set facts for node hosts
+  hosts: oo_nodes_to_config
+  roles:
+  - openshift_facts
+  tasks:
+  - openshift_facts:
+      role: "{{ item.role }}"
+      local_facts: "{{ item.local_facts }}"
+    with_items:
+      - role: common
+        local_facts:
+          hostname: "{{ openshift_hostname | default(None) }}"
+          public_hostname: "{{ openshift_public_hostname | default(None) }}"
+  - shell:
+      getent ahostsv4 {{ openshift.common.hostname }} | head -n 1 | awk '{ print $1 }'
+    register: lookupip
+    changed_when: false
+    failed_when: false
+  - pause:
+      prompt: "The hostname {{ openshift.common.hostname }} for {{ ansible_nodename }} doesn't resolve to an ip address owned by this host. Please set openshift_hostname variable to a hostname that when resolved on the host in question resolves to an IP address matching an interface on this host. This host will fail liveness checks for pods utilizing hostPorts, (a)bort or (c)ontinue"
+    when: lookupip.stdout not in ansible_all_ipv4_addresses
```

Comment:
Ok, I'll separate out that work and put it in a WIP PR encase anyone wants to pick it up where I've left it.